### PR TITLE
feat: filters in product listing

### PR DESCRIPTION
**Pull request by maitrakhatri**
# filter features
- Sort by price low to high, high to low
- filter by rating
- filter by category (checkbox)
- filter by price range
- clear all filters

### CODE REVIEW

File: gozo-stores/src/functions/functions.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        acc = acc.concat(productsArray);
+        return acc;
+    }
+
+    if(categories.length === 0) {
+        return products
+    }
+    else{
+        return categories.reduce(categoriedProductReducer, [])
+    }
```
Comment (Naveenchandar) [inline gozo-stores/src/functions/functions.js:14]: Suggestion: 
no need of else. You can just return **categories.reduce(categoriedProductReducer, [])**.
For checking the length, you can use not (!) operator.
Just a suggestion. Nothing wrong with the existing code. Please ignore it if not needed.

**Review by Naveenchandar — APPROVED** [review]
Clean code.
Nice folder structure.
If needed, you can rename the functions folder name as utils.

File: gozo-stores/src/components/FiltersSidebar.jsx [hunk 2]
```diff
@@ -1,50 +1,69 @@
...
-                        <span><input type="radio" name="rating" id="2+" />2+</span>
-                        <span><input type="radio" name="rating" id="2+" />1+</span>
+                        <span><input type="radio" name="rating" value="4" 
+                        onChange={(e) => dispatch({type: "RATING", payload: e.target.value})}/>4+</span>
+                        <span><input type="radio" name="rating" value="3" 
+                        onChange={(e) => dispatch({type: "RATING", payload: e.target.value})}/>3+</span>
+                        <span><input type="radio" name="rating" value="2" 
+                        onChange={(e) => dispatch({type: "RATING", payload: e.target.value})}/>2+</span>
+                        <span><input type="radio" name="rating" value="1"
+                        onChange={(e) => dispatch({type: "RATING", payload: e.target.value})}/>1+</span>
```
Comment (ranitasahaa1230) [inline gozo-stores/src/components/FiltersSidebar.jsx]: Use the map function for rating. These will reduce your lines of code too.

**Review by ranitasahaa1230 — APPROVED** [review]
Clean folder structure
Please add a screenshot or a GIF
Use map function to reduce your lines of code
Rest looks good to me

File: gozo-stores/src/functions/functions.js (hunk 1 above)
Comment (maitrakhatri) [reply gozo-stores/src/functions/functions.js:14]: Actually I didnt know I can do that, thanks !!

File: gozo-stores/src/components/FiltersSidebar.jsx (hunk 2 above)
Comment (maitrakhatri) [reply gozo-stores/src/components/FiltersSidebar.jsx]: yeah that's a good idea, as everything is same for them

**Pull request merged**